samples/matter/common/src/util/finite_map.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <map>
+
+namespace Nrf
+{
+/*
+   FiniteMap template container allows to wrap mappings between T1-type trivial key and T2-type value.
+   The size or the container is predefined at compilation time, therefore
+   the maximum number of stored elements must be well known. FiniteMap owns
+   inserted values, meaning that once the user inserts a value it will not longer
+   be valid outside of the container, i.e. FiniteMap cannot return stored object by copy.
+   FiniteMap's API offers basic operations like:
+     * inserting new values under provided key (Insert)
+     * erasing existing items under a given key (Erase) - this operation does not take care of the
+	   duplicated items,  meaning that the pointer under a given key is released unconditionally
```

Comment:
```suggestion
     * erasing existing items under a given key (Erase)
```
I think it's no longer true. This responsibility has been moved to the application.